src/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -291,10 +291,13 @@ private function normalizeDocumentType(string $html): string
      * Adds a Content-Type meta tag for the charset.
      *
      * This method also ensures that there is a HEAD element.
+     *
+     * @return non-empty-string
      */
     private function addContentTypeMetaTag(string $html): string
     {
         if ($this->hasContentTypeMetaTagInHead($html)) {
+            \assert($html !== '');
```

Comment:
This method is only called after the presence of a `<!DOCTYPE>` has been ensured, so will never be passed an empty string.  So I think we can add an `@param non-empty-string $html` annotation, and then probably won't need the `assert()`s.  Perhaps a description for the `$html` parameter to explain why an empty string is unexpected wouldn't go amiss.